--- netcf-0.1.7/src/drv_initscripts.c~	2010-09-23 14:13:13.000000000 +0200
+++ netcf-0.1.7/src/drv_initscripts.c	2011-03-28 20:11:43.092188473 +0200
@@ -52,14 +52,14 @@
     "/files/etc/sysconfig/network-scripts";
 
 static const char *const ifcfg_path =
-    "/files/etc/sysconfig/network-scripts/*";
+    "/files/etc/sysconfig/interfaces/*";
 
 /* Augeas should only load the files we are interested in */
 static const struct augeas_pv augeas_xfm_common_pv[] = {
     /* Ifcfg files */
     { "/augeas/load/Ifcfg/lens", "Sysconfig.lns" },
     { "/augeas/load/Ifcfg/incl",
-      "/etc/sysconfig/network-scripts/ifcfg-*" },
+      "/etc/sysconfig/interfaces/ifcfg-*" },
     { "/augeas/load/Ifcfg/excl[1]", "*~" },
     { "/augeas/load/Ifcfg/excl[2]", "*.bak" },
     { "/augeas/load/Ifcfg/excl[3]", "*.orig" },
